javascript/extractor/src/com/semmle/js/extractor/EnvironmentVariables.java

```diff
@@ -31,4 +34,12 @@ public static String getExtractorRoot() {
     }
     return env;
   }
+
+  public static String getScratchDir() {
+    String env = Env.systemEnv().get(CODEQL_EXTRACTOR_JAVASCRIPT_SCRATCH_DIR_ENV_VAR);
+    if (env == null) {
+      throw new UserError(CODEQL_EXTRACTOR_JAVASCRIPT_SCRATCH_DIR_ENV_VAR + " must be set");
```

Comment:
Will this work on LGTM? I don't think it sets that environment variable, does it? However, `LGTM_WORKSPACE` looks like a suitable alternative and is used by a few of the other autobuilders.

Reply:
Looking at the other AutoBuild scripts, it seems that `LGTM_WORKSPACE` is set to whatever value is already in `CODEQL_EXTRACTOR_LANG_SCRATCH_DIR`.

Reply:
I'm not sure I understand. Do you mean that LGTM sets `LGTM_WORKSPACE` to `CODEQL_EXTRACTOR_LANG_SCRATCH_DIR`? But who sets the latter? LGTM doesn't use CodeQL.